### DISCUSSION TITLE
I need help

### DISCUSSION THREAD

**Discussion by Alesker-Dadashov** [pymupdf/PyMuPDF#2768] [Looking for help]
Hi All, 
can you please help me with below code i cant move image4.jpg to right site of pdf page: 

`
from fitz import fitz
 
def add_footer(pdf_path, images, page_params, site="right"):
    try:
        pdf = fitz.open(pdf_path)
        page_count = pdf.page_count
 
        if page_count > 4:
            skip_pages = list(range(1, page_count - 3))
        else:
            skip_pages = []
 
        for i in range(page_count):
            if i in skip_pages:
                continue
 
            if i == 0:
                img_path1 = images[0]
                img_path2 = images[1]
                img1 = open(img_path1, "rb").read()
                img2 = open(img_path2, "rb").read()
            else:
                img_path = images[i +1% len(images)]
                img = open(img_path, "rb").read()
 
            page = pdf[i]
            if not page.is_wrapped:
                page.wrap_contents()
 
            if i < len(page_params):
                w, h, x1, x2 = page_params[i]
            else:
                w, h, x1, x2 = page_params[-1]
 
            if site == "right":
                rect = fitz.Rect(x1, x2, w, h)  # Full page size for right footer
                rect3 = fitz.Rect(x1 * w, x2 * h, w, h)  # Full page size for right footer
            else:
                rect = fitz.Rect(w * x1 * -1 * 0.94, h * x2, w, h)  # Full page size for left footer
 
            if i == 0:
                page.insert_image(rect3, stream=img1)
                # Adjust the coordinates for the second image as needed
                rect2 = fitz.Rect(0.80, 0.80,140.100 , 110.00)
                page.insert_image(rect2, stream=img2)
            else:
                page.insert_image(rect, stream=img)
 
        pdf.save("output.pdf")
        pdf.close()
 
        print("Footer added successfully.")
 
    except Exception as e:
        print(f"An error occurred: {str(e)}")
 
# Example usage with different images, positions, and rotations for pages 0, 1, and 2
page_params = [
    (200.0, 508.00, 0.110, 0.133),  # Parameters for page 0
    (295.00, 750.00, 0.400, 0.400),    # Parameters for page 1
    (295.00, 795.00, 0.400, 0.400), # Parameters for page 2
    (150.00, 400.00, 100.00, 100.00)
]
 
# Images for page 0, 1, and 2
images = ["image2.jpg", "image1.jpg", "image5.jpg", "image5.jpg", "image4.jpg"]
 
add_footer("file.pdf", images, page_params, site="right")
`

**Comment by JorjMcKie** [comment]
Your code is hard to understand, but I noted one thing: The definition for `fitz.Rect(x0, y0, x1, y1)` consists of coordinates of the top-left (x0, y0) and bottom-right (x1, y1)  points.
Width and height as you use it, are not a part of this.
